examples/clients/clients.bzl

```diff
@@ -0,0 +1,53 @@
+def swagger_codegen(name, spec, package_name, outputs, out_dir=".", **kwarg):
+  out_dir_spec = "$(@D)"
+  if out_dir != ".":
+    out_dir_spec += "/" + out_dir
+    for out in outputs:
+      if not out.startswith(out_dir + "/"):
+        fail("%s must reside in the out_dir %s" % (out, out_dir), "outputs")
+
+  native.genrule(
+      name = name,
+      srcs = [spec],
+      outs = outputs,
+      cmd = " ".join([
+          "$(location //examples/clients:swagger-codegen-cli)",
```

Comment:
Github is being... weird. I typed out a whole comment in here but it wasn't included.

Here is approximately what I wrote:

What would you think about using a temp dir to generate these in along with a `.swagger-ignore-codegen` (or set of rm statements) and then a `mv` from the temp to the final `out_dir_spec` followed by an `rmdir` on the temp dir. This has the nice benefit of making it a compile time error if a new file is generated but not enumerated in the srcs section.